Apps/phdetectionondemand/detectionondemand_connector.py

```diff
@@ -0,0 +1,535 @@
+# Python 3 Compatibility imports
+from __future__ import print_function, unicode_literals
+
+# Phantom App imports
+import phantom.app as phantom
+from phantom.vault import Vault
+from phantom.base_connector import BaseConnector
+from phantom.action_result import ActionResult
+
+# Usage of the consts file is recommended
+from detectionondemand_consts import *
+import requests
+import time
+import json
+import sys
+from bs4 import BeautifulSoup, UnicodeDammit
+
+
+class RetVal(tuple):
+
+    def __new__(cls, val1, val2=None):
+        return tuple.__new__(RetVal, (val1, val2))
+
+
+class DetectionOnDemandConnector(BaseConnector):
+
+    def __init__(self):
+
+        # Call the BaseConnectors init first
+        super(DetectionOnDemandConnector, self).__init__()
+
+        self._state = None
+        self._base_url = None
+        self._api_token = None
+
+    def _validate_integer(self, action_result, parameter, key):
+        if parameter is not None:
+            try:
+                if not float(parameter).is_integer():
+                    return action_result.set_status(phantom.APP_ERROR, "Please provide a valid integer value in the {}".format(key)), None
```

Comment:
ERR vs ERROR consts consistency

Reply:
Should it be phantom.APP_ERR?

Reply:
Apologies, it is correct the way you have it.